src/loki/search.cc

```diff
@@ -476,11 +473,14 @@ struct bin_handler_t {
 
     // if the inbound reach is not 0 and the outbound reach is not 0 and the opposing edge is not
     // filtered then the reaches of both edges are the same
+
     const DirectedEdge* opp_edge = nullptr;
-    if (reach.outbound > 0 && reach.inbound > 0 && (opp_edge = reader.GetOpposingEdge(edge, tile)) &&
-        edge_filter(opp_edge) > 0.f)
-      directed_reaches[opp_edge] = reach;
+    const auto opp_edgeid = reader.GetOpposingEdgeId(edge_id, opp_edge, tile);
```

Comment:
we want this to be further down in the terms for the following `if` statement because its heavier than the other checks that happen before it. but always doing  it regardless you make sure its the worst performance possible. you should put it back inside the `if` statement below